src/fsstorage.cc

```diff
@@ -0,0 +1,179 @@
+  storeTlsCredsCommon(ca_path, cert_path, pkey_path, ca, cert, pkey);
+}
+
+void FSStorage::storeTlsCreds(const std::string& ca, const std::string& cert, const std::string& pkey) {
+  boost::filesystem::path ca_path = config_.tls.certificates_directory / config_.tls.ca_file;
+  boost::filesystem::path cert_path = config_.tls.certificates_directory / config_.tls.client_certificate;
+  boost::filesystem::path pkey_path = config_.tls.certificates_directory / config_.tls.pkey_file;
+  storeTlsCredsCommon(ca_path, cert_path, pkey_path, ca, cert, pkey);
+}
+
+bool FSStorage::loadTlsCredsCommon(const boost::filesystem::path& ca_path, const boost::filesystem::path& cert_path,
+                                   const boost::filesystem::path& pkey_path, std::string* ca, std::string* cert,
+                                   std::string* pkey) {
+  if (!boost::filesystem::exists(ca_path) || !boost::filesystem::exists(cert_path) ||
+      !boost::filesystem::exists(pkey_path))
+    return false;
+
+  *ca = Utils::readFile(ca_path.native());
+  *cert = Utils::readFile(cert_path.native());
+  *pkey = Utils::readFile(pkey_path.native());
+
+  return false;
+}
+
+bool FSStorage::loadBootstrapTlsCreds(std::string* ca, std::string* cert, std::string* pkey) {
+  boost::filesystem::path ca_path = config_.tls.certificates_directory / "bootstrap_ca.pem";
+  boost::filesystem::path cert_path = config_.tls.certificates_directory / "bootstrap_cert.pem";
+  boost::filesystem::path pkey_path = config_.tls.certificates_directory / "bootstrap_pkey.pem";
+  return loadTlsCredsCommon(ca_path, cert_path, pkey_path, ca, cert, pkey);
+}
+
+bool FSStorage::loadTlsCreds(std::string* ca, std::string* cert, std::string* pkey) {
+  boost::filesystem::path ca_path = config_.tls.certificates_directory / config_.tls.ca_file;
+  boost::filesystem::path cert_path = config_.tls.certificates_directory / config_.tls.client_certificate;
+  boost::filesystem::path pkey_path = config_.tls.certificates_directory / config_.tls.pkey_file;
+  return loadTlsCredsCommon(ca_path, cert_path, pkey_path, ca, cert, pkey);
+}
+
+void FSStorage::storeMetadata(const Uptane::MetaPack& metadata) {
+  boost::filesystem::path image_path = config_.uptane.metadata_path / "repo";
+  boost::filesystem::path director_path = config_.uptane.metadata_path / "director";
+
+  Utils::writeFile((director_path / "root.json").native(), metadata.director_root.toJson());
+  Utils::writeFile((director_path / "targets.json").native(), metadata.director_targets.toJson());
+  Utils::writeFile((image_path / "root.json").native(), metadata.image_root.toJson());
+  Utils::writeFile((image_path / "targets.json").native(), metadata.image_targets.toJson());
+  Utils::writeFile((image_path / "timestamp.json").native(), metadata.image_timestamp.toJson());
+  Utils::writeFile((image_path / "snapshot.json").native(), metadata.image_snapshot.toJson());
+  sync();
+}
+
+bool FSStorage::loadMetadata(Uptane::MetaPack* metadata) {
```

Comment:
I think loadMetadata and storeMetadata should be memebres of Uptane class. 
"repo"  and "director" might have another names and only Uptane knows them